TestWeightedPath.test_weight_functions fails on i586

I'm packaging networkx 2.5 for openSUSE Tumbleweed but when building it for i586 the algorithms.shortest_paths.tests.test_weighted.TestWeightedPath.test_weight_functions fail:

```
=================================== FAILURES ===================================
____________________ TestWeightedPath.test_weight_functions ____________________

self = <networkx.algorithms.shortest_paths.tests.test_weighted.TestWeightedPath object at 0xa3db1118>

    def test_weight_functions(self):
        def heuristic(*z):
            return hash(z)
    
        def getpath(pred, v, s):
            return [v] if v == s else getpath(pred, pred[v], s) + [v] 
    
        def goldberg_radzik(g, s, t, weight="weight"):
            pred, dist = nx.goldberg_radzik(g, s, weight=weight)
            dist = dist[t]
            return dist, getpath(pred, t, s)
    
        def astar(g, s, t, weight="weight"):
            path = nx.astar_path(g, s, t, heuristic, weight=weight)
            dist = nx.astar_path_length(g, s, t, heuristic, weight=weight)
            return dist, path
    
        def vlp(G, s, t, l, F, w):
            res = F(G, s, t, weight=w)
            validate_length_path(G, s, t, l, *res, weight=w)
    
        G = self.cycle
        s = 6    
        t = 4    
        path = [6] + list(range(t + 1))
    
        def weight(u, v, _):
            return 1 + v ** 2 
    
        length = sum(weight(u, v, None) for u, v in pairwise(path))
        vlp(G, s, t, length, nx.bidirectional_dijkstra, weight)
        vlp(G, s, t, length, nx.single_source_dijkstra, weight)
        vlp(G, s, t, length, nx.single_source_bellman_ford, weight)
        vlp(G, s, t, length, goldberg_radzik, weight)
>       vlp(G, s, t, length, astar, weight)

networkx/algorithms/shortest_paths/tests/test_weighted.py:233: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  
networkx/algorithms/shortest_paths/tests/test_weighted.py:218: in vlp
    validate_length_path(G, s, t, l, *res, weight=w)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

G = <networkx.classes.graph.Graph object at 0xa3db1370>, s = 6, t = 4 
soln_len = 35, length = 43, path = [6, 5, 4]
weight = <function TestWeightedPath.test_weight_functions.<locals>.weight at 0xa3da4268>

    def validate_length_path(G, s, t, soln_len, length, path, weight="weight"):
>       assert soln_len == length
E       assert 35 == 43 
E         +35
E         -43

networkx/algorithms/shortest_paths/tests/test_weighted.py:29: AssertionError
=========================== short test summary info ============================
FAILED networkx/algorithms/shortest_paths/tests/test_weighted.py::TestWeightedPath::test_weight_functions
============ 1 failed, 3989 passed, 40 skipped in 211.14s (0:03:31) ============
```

All tests work fine on x86_64, aarch64, ppc64 and ppc64le . You can see the full build log at [the openSUSE build service](https://build.opensuse.org/public/build/home:alarrosa:branches:devel:languages:python:numeric/openSUSE_Tumbleweed/i586/python-networkx/_log), and the build logs of the other architectures builds by clicking on the `succeeded` or `failed` build result status [here](https://build.opensuse.org/package/show/home:alarrosa:branches:devel:languages:python:numeric/python-networkx).

If there's anything I can test, please tell me.

--- a/networkx/algorithms/shortest_paths/tests/test_weighted.py
+++ b/networkx/algorithms/shortest_paths/tests/test_weighted.py
@@ -198,7 +198,7 @@ class TestWeightedPath(WeightedTestBase):
 
     def test_weight_functions(self):
         def heuristic(*z):
-            return hash(z)
+            return sum(val ** 2 for val in z)
 
         def getpath(pred, v, s):
             return [v] if v == s else getpath(pred, pred[v], s) + [v]

